Tasks/ArchiveFilesV2/utils.ts

```diff
@@ -0,0 +1,17 @@
+import tl = require("vsts-task-lib/task");
+
+export function reportArchivePlan(files: string[], max: number=10) {
+    var plan: string[] = [];
+    plan.push(tl.loc('FoundNFiles', files.length));
+    if (files.length > 0) {
+        for (var i = 0; i < files.length; i++) {
+            plan.push(tl.loc('ArchivingFile',files[i]));
+            let remaining = files.length - (i+1);
+            if (i >= max-1 && remaining > 0) {
```

Comment:
Instead of having this logic in the for can't we just do the for loop over the files and add up to max, then after the for is done have an if statement to check `if files.length > max` then we add `MoreFiles` log? Would be easier to understand.